lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -168,13 +168,18 @@ class AwsCompileCloudFrontEvents {
             // Retain Lambda@Edge functions to avoid issues when removing the CloudFormation stack
             Object.assign(Resources[lambdaFunctionLogicalId], { DeletionPolicy: 'Retain' });
 
-            const lambdaVersionLogicalId = _.findKey(Resources, {
-              Type: 'AWS::Lambda::Version',
-              Properties: {
-                FunctionName: {
-                  Ref: this.provider.naming.getLambdaLogicalId(functionName),
-                },
-              },
+            const lambdaVersionLogicalId = Object.keys(Resources).find(key => {
+              const r = Resources[key];
+              if (r.Type !== 'AWS::Lambda::Version') return false;
+              if (!r.Properties) return false;
+              if (!r.Properties.FunctionName) return false;
+              if (
+                r.Properties.FunctionName.Ref !==
+                this.provider.naming.getLambdaLogicalId(functionName)
+              ) {
+                return false;
+              }
+              return true;
```

Comment:
I think it'll be nicer to user `_.get` here, as

```javascript
return _.get(resource, 'Properties.FunctionName.Ref') === lambdaFunctionLogicalId
```

Note: I think it's still good to rely on `lodash` in justified cases. Like here, sticking to `_.get`, will make logic definitely more readable

Reply:
I understand this, but I see in some other functions there are native checks like what I did here. It's kind of inconsistent I think.

Reply:
We have many bad patterns across project, through which we gradually recover. I think our decisions should not be necessarily influenced by how other parts were written, but what's best in terms for code quality (readability, future maintenance etc.)

What in your opinion is cleanest form to express this logic?

Reply:
I'd personally change the 2 lines 174 and 175 into 1 line like this:
```javascript
if (!resource.Properties || !resource.Properties.FunctionName) return false;
```
Because sooner or later `_.get` would be replaced.

Reply:
> Because sooner or later _.get would be replaced.

If `_.get` would be replaced, it'll be either with counterpart coming from other utils library, or native [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (once support for Node.js v12 is dropped).

So it's good to write code which in future could be replaced easily (in place) into:

```javascript
return resource?.Properties?.FunctionName?.Ref === lambdaFunctionLogicalId
```

Note that #7747 is not about replacing lodash entirely. It's just about replacing parts which can be written with native syntax, without sacrificing readability